app/src/services/sync.js

```diff
@@ -329,15 +329,18 @@ const service = {
       // S3 Tags
       const s3Tags = toLowerKeys(s3TagsForVersion?.TagSet ?? []);
 
-      // Ensure `coms-id` tag exists on this version in S3
-      if (s3Tags.length < 10 && !s3Tags.find(s3T => s3T.key === 'coms-id')) {
+      // Add coms-id tag to latest version in S3 if not already present
+      // note: For a sync job the _deriveObjectId() function will have already added
+      // the coms-id to latest version.
+      if (version.isLatest && s3Tags.length < 10 && !s3Tags.find(s3T => s3T.key === 'coms-id')) {
```

Comment:
Just realizing that we're pulling isLatest data from COMS lineage, and not potentially from S3. We might need to do one extra storageService call to get the current version state and check the isLatest from there instead of a potentially derived coms one?

Reply:
as discussed, this is an edge case and acceptable risk we can consider fixing as required (for example, if we re-use SyncTags() outside of the complete object sync job.